tests/forte/processors/remote_processor_test.py

```diff
@@ -111,12 +119,23 @@ def test_remote_processor(self, input_output_pair):
 
         # Configure RemoteProcessor into test mode
         remote_processor: RemoteProcessor = RemoteProcessor()
-        remote_processor.set_test_mode(serve_pl._remote_service_app)
+        remote_processor.set_test_mode(
+            serve_pl._remote_service_app(name=service_name)
+        )
 
         # Build test pipeline
         test_pl: Pipeline[DataPack] = Pipeline[DataPack]()
         test_pl.set_reader(StringReader())
-        test_pl.add(remote_processor)
+        test_pl.add(
+            remote_processor,
```

Comment:
I just realize here the remote service can only function correctly if we use a reader that can take a data pack (RawDataDeserializeReader in this case). If our remote processor is configured so that the reader is something else like `StringReader`, then it may interpret the input wrong. This is something we could check during validation. Because we are building a `DataPack` pipeline, then it should check whether the remote service supports `DataPack`. We can do the following:

1. During `serve()`, we can have a configuration `input_format="string"`, which means the input format defaults to `string` (or other default values you find appropriate). For the remote service to support "DataPack", we need to set the `input_format` to be `DataPack` (in the future we may support `MultiPack` too).  For the time being, we could support these two values. However, we do not have a way for the `Reader` to check this input. This could be a potential feature that's useful, we might do it someday.
2. During `add(remote_processor)`, we should check whether the output from the previous step matches the remote processor configuration. Unfortunately, in Python, we cannot reliably find the output type of the previous PipelineComponent, hence we simply add this check as part of the validation configuration.

We should also group the validation settings into the same group, like the following:
```
config={               
   "validation": {
       "do_validation": True,                
      "input_format": "DataPack",
      "expected_name": service_name,                
       "expected_records": {
            "Token": ["1", "2"], 
            "Document": ["2"]}  
       },
   }
```